fuel-core/src/cli/run/p2p.rs

```diff
@@ -133,10 +136,11 @@ pub struct P2pArgs {
 }
 
 impl P2pArgs {
-    pub fn into_config(self, chain_config: &ChainConfig) -> anyhow::Result<P2PConfig> {
-        let checksum = *fuel_crypto::Hasher::default()
-            .chain(bincode::serialize(chain_config)?)
-            .finalize();
+    pub fn into_config(
+        self,
+        chain_config: &mut ChainConfig,
+    ) -> anyhow::Result<P2PConfig> {
+        let checksum = chain_config.root()?.into();
```

Comment:
`Genesis::root` contains all information

Reply:
We can solve it in this https://github.com/FuelLabs/fuel-core/pull/828 way, for example